Mathlib/Topology/Sheaves/Stalks.lean

```diff
@@ -229,71 +223,112 @@ section stalkPullback
 
 /-- The morphism `ℱ_{f x} ⟶ (f⁻¹ℱ)ₓ` that factors through `(f_*f⁻¹ℱ)_{f x}`. -/
 def stalkPullbackHom (f : X ⟶ Y) (F : Y.Presheaf C) (x : X) :
-    F.stalk (f x) ⟶ (pullbackObj f F).stalk x :=
+    F.stalk (f x) ⟶ ((pullback C f).obj F).stalk x :=
   (stalkFunctor _ (f x)).map ((pushforwardPullbackAdjunction C f).unit.app F) ≫
     stalkPushforward _ _ _ x
 set_option linter.uppercaseLean3 false in
 #align Top.presheaf.stalk_pullback_hom TopCat.Presheaf.stalkPullbackHom
 
+@[reassoc (attr := simp)]
+lemma germ_stalkPullbackHom
+    (f : X ⟶ Y) (F : Y.Presheaf C) (x : X) (U : Opens Y) (hU : f x ∈ U) :
+    F.germ ⟨f x, hU⟩ ≫ stalkPullbackHom C f F x =
+      ((pushforwardPullbackAdjunction C f).unit.app F).app _ ≫
+        ((pullback C f).obj F).germ ⟨x, show x ∈ (Opens.map f).obj U from hU⟩ := by
+  simp [stalkPullbackHom, germ, stalkFunctor, stalkPushforward]
+
 /-- The morphism `(f⁻¹ℱ)(U) ⟶ ℱ_{f(x)}` for some `U ∋ x`. -/
 def germToPullbackStalk (f : X ⟶ Y) (F : Y.Presheaf C) (U : Opens X) (x : U) :
-    (pullbackObj f F).obj (op U) ⟶ F.stalk ((f : X → Y) (x : X)) :=
-  colimit.desc (Lan.diagram (Opens.map f).op F (op U))
+    ((pullback C f).obj F).obj (op U) ⟶ F.stalk ((f : X → Y) (x : X)) :=
+  ((Opens.map f).op.isPointwiseLeftKanExtensionLanUnit F (op U)).desc
     { pt := F.stalk ((f : X → Y) (x : X))
       ι :=
         { app := fun V => F.germ ⟨((f : X → Y) (x : X)), V.hom.unop.le x.2⟩
           naturality := fun _ _ i => by erw [Category.comp_id]; exact F.germ_res i.left.unop _ } }
 set_option linter.uppercaseLean3 false in
 #align Top.presheaf.germ_to_pullback_stalk TopCat.Presheaf.germToPullbackStalk
 
+variable {C} in
+@[ext]
+lemma pullback_obj_obj_ext {Z : C} {f : X ⟶ Y} {F : Y.Presheaf C} (U : (Opens X)ᵒᵖ)
```

Comment:
Why did this file become quite a bit longer? Is this hinting at a hole in the new Kan extension API?

Reply:
I will try to investigate the issue closer.